src/webgpu/api/validation/validation_test.ts

```diff
@@ -280,26 +281,39 @@ export class ValidationTest extends GPUTest {
     }
   }
 
-  expectValidationError(fn: Function, shouldError: boolean = true): void {
+  /**
+   * Expect a validation error inside the callback.
+   *
+   * Awaiting the returned promise is only necessary if you want to await on the completion
+   * of an async callback. The validation error check will still happen "eventually".
+   *
+   * Tests should always do just one WebGPU call in the callback, to make sure that's what's tested.
+   */
+  expectValidationError(fn: () => unknown, shouldError: boolean = true): void {
```

Comment:
```suggestion
  expectValidationError(fn: () => undefined, shouldError: boolean = true): void {
```

if we have the assertion below at [1] on the return value, can we just say the type here should be undefined?

Reply:
Unfortunately there is a weird quirk in TypeScript here. `() => {}` is not a valid `() => undefined` (because `void` is not assignable to `undefined`) but `async () => {}` and `() => 3` *are* valid `() => void`s (even though `Promise<void>` and `number` are not assignable to `void`). So AFAICT no matter what we put here we can't get better type safety without requiring `return undefined;` in all of the user callbacks.

I'll change it to void, it should be equivalent but more self-documenting.